core/tee/tee_rpmb_fs.c

```diff
@@ -2433,35 +2442,87 @@ static TEE_Result rpmb_fs_write_primitive(struct rpmb_file_handle *fh,
 		 * File must be extended, or update cannot be atomic: allocate,
 		 * read, update, write.
 		 */
+		uintptr_t old_fat = fh->fat_entry.start_address;
```

Comment:
The code would be easier to follow if most of what's in this else was moved into a helper function.

Reply:
lot of variable to track the 2 sources and 1 dest :(

Reply:
``` C
static TEE_Result update_write_helper(struct rpmb_file_handle *fh,
                                      size_t pos, const void *buf,
                                      size_t size, uintptr_t new_fat,
                                      size_t new_size)
```
seems manageable.

Reply:
oh, i single helper function. yes.
sorry, I misread your comment. will do.